### PR TITLE
Update Cluster API GCP provider presubmits

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api-provider-gcp/cluster-api-provider-gcp-presubmits.yaml [hunk 1]
```diff
@@ -1,61 +1,28 @@
...
-        - "--repo=sigs.k8s.io/$(REPO_NAME)=$(PULL_REFS)"
-        - "--root=/go/src"
-        - "--upload=gs://kubernetes-jenkins/pr-logs"
-        - "--scenario=execute"
-        - "--"
-        - "./scripts/ci-build.sh"
-    annotations:
-      testgrid-dashboards: sig-cluster-lifecycle-cluster-api-provider-gcp
-      testgrid-tab-name: pr-build
-  - name: pull-cluster-api-provider-gcp-make
```
Comment (vincepri) [inline config/jobs/kubernetes-sigs/cluster-api-provider-gcp/cluster-api-provider-gcp-presubmits.yaml]: The make target isn't actually doing anything differently than build, so we remove it

**Comment by vincepri** [comment]
/assign @cblecker 

for the squash change

File: config/jobs/kubernetes-sigs/cluster-api-provider-gcp/cluster-api-provider-gcp-presubmits.yaml (hunk 1 above)
Comment (detiber) [reply config/jobs/kubernetes-sigs/cluster-api-provider-gcp/cluster-api-provider-gcp-presubmits.yaml]: It looks like scripts/ci-make.sh is calling `make docker-build` and scripts/ci-build.sh is calling `make manager clusterctl`, so I'm not sure these should be de-duped currently.

Comment (vincepri) [reply config/jobs/kubernetes-sigs/cluster-api-provider-gcp/cluster-api-provider-gcp-presubmits.yaml]: Not sure what I was looking at, fixed!

**Comment by cblecker** [comment]
My approve will count for both.. I'd like to let the cluster-provider-gcp owners have a look first.
/cc @roberthbailey @justinsb @krousey @maisem

**Comment by detiber** [comment]
I'm thinking that OWNER list might need a bit of a refresh :joy:

**Comment by vincepri** [comment]
I can update the file as part of this PR, it should match these, correct? https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/master/OWNERS_ALIASES#L18-L23

**Comment by detiber** [comment]
@vincepri correct, it should match the maintainers for the associated repo.

**Comment by vincepri** [comment]
Updated!

**Comment by roberthbailey** [comment]
thanks for updating the owners file!

**Pull request merged**